<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ScientificCalculator</class>
 <widget class="QMainWindow" name="ScientificCalculator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Scientific Calculator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLabel" name="processLabel">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>30</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="display">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>50</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>20</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <!-- Scientific Functions Row 1 -->
      <item row="0" column="0">
       <widget class="QPushButton" name="buttonSin">
        <property name="text">
         <string>sin</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="buttonCos">
        <property name="text">
         <string>cos</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="buttonTan">
        <property name="text">
         <string>tan</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="buttonPi">
        <property name="text">
         <string>π</string>
        </property>
       </widget>
      </item>
      <!-- Scientific Functions Row 2 -->
      <item row="1" column="0">
       <widget class="QPushButton" name="buttonLog">
        <property name="text">
         <string>log</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="buttonLn">
        <property name="text">
         <string>ln</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="buttonExp">
        <property name="text">
         <string>exp</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QPushButton" name="buttonE">
        <property name="text">
         <string>e</string>
        </property>
       </widget>
      </item>
      <!-- Scientific Functions Row 3 -->
      <item row="2" column="0">
       <widget class="QPushButton" name="buttonPower">
        <property name="text">
         <string>x^y</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="buttonSquare">
        <property name="text">
         <string>x²</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="buttonSqrt">
        <property name="text">
         <string>√</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QPushButton" name="buttonFactorial">
        <property name="text">
         <string>n!</string>
        </property>
       </widget>
      </item>
      <!-- Original Calculator Buttons -->
      <!-- [Previous button layout remains the same] -->
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
</ui>